hack/bin/integration-setup-federation.sh

```diff
@@ -1,19 +1,56 @@
 #!/usr/bin/env bash
 
-USAGE="Usage: $0"
+DIR="$(cd "$(dirname "${BASH_SOURCE[0]}")" && pwd)"
+TOP_LEVEL="$DIR/../.."
+export NAMESPACE=${NAMESPACE:-test-integration}
+CHARTS_DIR="${TOP_LEVEL}/.local/charts"
+
+. "$DIR/helm_overrides.sh"
+${DIR}/integration-cleanup.sh
+
+# FUTUREWORK explore: have helmfile do the interpolation (and skip the "make charts" step) https://wearezeta.atlassian.net/browse/SQPIT-722
+#
+# FUTUREWORK: get rid of wrapper charts, use helmfile for pinning. Then we may not need the recursive update hack anymore: https://wearezeta.atlassian.net/browse/SQPIT-721
+#
+# Sadly, even with helmfile, we still need to use use this recursive update
+# script beforehand on all relevant charts to download the nested dependencies
+# (e.g. cassandra from underneath databases-ephemeral)
+echo "updating recursive dependencies ..."
+charts=(fake-aws databases-ephemeral wire-server nginx-ingress-controller nginx-ingress-services)
+for chart in "${charts[@]}"; do
+    "$DIR/update.sh" "$CHARTS_DIR/$chart"
+done
 
 set -e
```

Comment:
This should be on top, also while we're at it, we should make it `set -euo pipefail`

Reply:
`man bash` says `-o errexit` is same as `-e`. I am curious why it is frowned upon.